samples/net/updatehub/README.rst

```diff
@@ -0,0 +1,240 @@
+UpdateHub sample
+################
+
+Overview
+********
+
+UpdateHub is an enterprise-grade solution which makes it simple to remotely
+update all your Linux-based devices in the field. It handles all aspects
+related to sending Firmware Over-the-Air (FOTA) updates with maximum
+security and efficiency, while you focus on adding value to your product.
+
+This sample shows how to use UpdateHub in both a polling and manual update
+mode.
+
+Polling mode runs automatically on a predefined period, probing the server
+for updates and installing them without requiring user intervention. You
+can access the sample source code for this mode updatehub_polling.
+
+Manual mode requires the user to call the server probe and then, if there is
+an available update, also requires the user to decide if it is appropriate to
+update now or later. You can access the sample source code for this mode
+updatehub_manual.
+
+Caveats
+*******
+
+* The Zephyr port of ``UpdateHub`` is configured to run on a Freedom-K64F
+  MCU by default. The application should build and run for other platforms
+  with offer support internet connection. Some platforms need some modification.
+  Overlay files would be needed to support BLE 6lowpan, 802.15.4 or OpenThread
+  configurations as well as the understanding that most other connectivity
+  options would require an edge gateway of some sort (Border Router, etc).
+
+* The MCUboot bootloader is required for ``UpdateHub`` to function
+  properly. More information about the Device Firmware Upgrade subsystem and
+  MCUboot can be found in :ref:`mcuboot`.
+
+Building and Running
+********************
+
+The below steps describe how to build and run the ``UpdateHub`` sample in
+Zephyr. Where examples are given, it is assumed the sample is being built for
+the Freedom-K64F Development Kit (``BOARD=fdrm_k64f``).
+
+Step 1: Build MCUboot
+=====================
+
+Build MCUboot by following the instructions in the :ref:`mcuboot` documentation
+page.
+
+Step 2: Flash MCUboot
+=====================
+
+Flash the resulting image file to the 0x0 address of the flash memory. This can
+be done in multiple ways, but the most common ones would be using make or ninja:
+
+.. code-block:: console
+
+   make flash
+   # or
+   ninja flash
+
+Step 3: Start the updatehub-ce-server
+=====================================
+
+By default, the updatehub application is set to start on the official server.
+For more details on how to use the official server please refer to the
+documentation on `updatehub.io`_.
+
+.. _updatehub.io: https://updatehub.io
```

Comment:
You could collect these links at the end of the document. It is easier to read the file if they are not cluttering the text.